rotkehlchen/chain/evm/decoding/morpho/utils.py

```diff
@@ -78,58 +77,45 @@ def _query_morpho_vaults_api() -> list[dict[str, Any]] | None:
     return all_vaults
 
 
-def query_morpho_vaults(database: 'DBHandler') -> None:
-    """Query list of Morpho vaults and add the vault tokens to the global database."""
-    with GlobalDBHandler().conn.read_ctx() as cursor:
-        last_vault_count = globaldb_get_unique_cache_value(
-            cursor=cursor,
-            key_parts=(CacheType.MORPHO_VAULTS,),
-        )
-
-    if (vault_list := _query_morpho_vaults_api()) is None:
-        _update_cache_timestamp()  # Update timestamp to prevent repeated errors.
-        return
+def _process_morpho_vault(database: 'DBHandler', vault: dict[str, Any]) -> None:
```

Comment:
Isn't this also further abstractable? Don't the other vaults all do the same thing too with just protocol name and protocol being variable?

Reply:
Unfortunately no, the vault processing can't be abstracted very well. The vault data is not in the same format from the two apis, so the keys used when getting data from the vault dict are different for a number of things. Also, for curve lending vaults there's some extra logic to determine the chain id, as well as a couple addresses that are saved to the cache. So its not just the protocol and display name that are different.